src/python/tests_extended/test_dft_based.py

```diff
@@ -0,0 +1,662 @@
+
+        xf = ToKeyImputer()
+        mlnetresult = xf.fit_transform(df)
+
+        onnx_path = get_tmp_file('.onnx')
+        onnx_json_path = get_tmp_file('.onnx.json')
+        xf.export_to_onnx(onnx_path,
+                            'com.microsoft.ml',
+                            dst_json=onnx_json_path,
+                            onnx_version='Stable')
+        df_tool = DFT(onnx_path)
+        result_ort = df_tool.execute(df, [])
+        
+        if len(result_ort.columns) != len(mlnetresult.columns):
+            raise RuntimeError("ERROR: The ORT output does not contain the same number of columns as ML.NET.")
+        col_tuples = list(zip(mlnetresult.columns[0:],
+                              result_ort.columns[0:]))
+        for col_tuple in col_tuples:
+            try:
+                col_mlnet = mlnetresult.loc[:, col_tuple[0]]
+                col_ort = result_ort.loc[:, col_tuple[1]]
+                check_kwargs = {
+                    'check_names': False,
+                    'check_exact': True,
+                    'check_dtype': True,
+                    'check_less_precise': True
+                }
+                pd.testing.assert_series_equal(col_mlnet, col_ort, **check_kwargs)
+            except Exception as e:
+                print(e)
+                raise RuntimeError("ERROR: OnnxRunner result does not match ML.NET result.")
+
+    def test_tokey_2col_double_string(self):
+        
+        df = pd.DataFrame(data=dict(
+            data1=[1.0, 1.0, 1.0, 2.0],
+            data2=["two", "two", "three", "two"],
+        ))
+
+        xf = ToKeyImputer()
+        mlnetresult = xf.fit_transform(df)
+
+        onnx_path = get_tmp_file('.onnx')
+        onnx_json_path = get_tmp_file('.onnx.json')
+        xf.export_to_onnx(onnx_path,
+                            'com.microsoft.ml',
+                            dst_json=onnx_json_path,
+                            onnx_version='Stable')
+        df_tool = DFT(onnx_path)
+        result_ort = df_tool.execute(df, [])
+        
+        if len(result_ort.columns) != len(mlnetresult.columns):
+            raise RuntimeError("ERROR: The ORT output does not contain the same number of columns as ML.NET.")
+        col_tuples = list(zip(mlnetresult.columns[0:],
+                              result_ort.columns[0:]))
+        for col_tuple in col_tuples:
+            try:
+                col_mlnet = mlnetresult.loc[:, col_tuple[0]]
+                col_ort = result_ort.loc[:, col_tuple[1]]
+                check_kwargs = {
+                    'check_names': False,
+                    'check_exact': True,
+                    'check_dtype': True,
+                    'check_less_precise': True
+                }
+                pd.testing.assert_series_equal(col_mlnet, col_ort, **check_kwargs)
+            except Exception as e:
+                print(e)
+                raise RuntimeError("ERROR: OnnxRunner result does not match ML.NET result.")
```

Comment:
If you refactor, then ideally the whole test should look like: df = {some data};  run_tokey_test(df) . So then we could just focus on testing various df